Nos/Views/New Note/FileStorageAPI.swift

```diff
@@ -0,0 +1,34 @@
+import Foundation
+import SwiftUI
+
+struct AttachedFile {
+    var image: UIImage
+}
+
+protocol FileStorageAPI {
+    func upload(file: AttachedFile) async throws -> URL
+}
+
+class NostrBuildFileStorageAPI: FileStorageAPI {
+    var uploadURL = URL(string: "https://nostr.build/api/upload/ios.php")!
+    
+    func upload(file: AttachedFile) async throws -> URL {
+        let boundary = UUID().uuidString
+        let fileName = "file.jpg"
+        let paramName = "fileToUpload"
+        var request = URLRequest(url: uploadURL)
+        request.httpMethod = "POST"
+        request.setValue("multipart/form-data; boundary=\(boundary)", forHTTPHeaderField: "Content-Type")
+        
+        var data = Data()
+        data.append("\r\n--\(boundary)\r\n".data(using: .utf8)!)
+        data.append("Content-Disposition: form-data; name=\"\(paramName)\"; filename=\"\(fileName)\"\r\n".data(using: .utf8)!)
+        data.append("Content-Type: image/jpg\r\n\r\n".data(using: .utf8)!)
+        data.append(file.image.jpegData(compressionQuality: 85)!)
+        data.append("\r\n--\(boundary)--\r\n".data(using: .utf8)!)
+        
+        let (responseData, _) = try await URLSession.shared.upload(for: request, from: data)
+        let url = try JSONSerialization.jsonObject(with: responseData, options: .allowFragments) as? String
+        return URL(string: url!)!
```

Comment:
We might not need to do JSON deserialization here. If the reponseData is just a String then you can use `String(data: responseData, encoding: .utf8)`.

And let's not crash if the server returns something that isn't a string or a valid URL. In that case we can throw an error (might need to make a new FileStorageAPIError type to throw)

Reply:
It is not just a string, it is specifically a JSON string wrapped in `""` as specified by [ECMA-404](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf).

Reply:
Both are correct somehow, but I prefer filip way as it is more declarative of the fact that it is a json response.
But we need to address the two ! at line 32 and throw errors if any of those cannot be unwrapped.

Reply:
Nope, the approaches are not equivalent. The string approach:
- won't handle `"` around the string
- won't decode escape sequences present in the JSON string

The string approach wouldn't function the same way even if we striped the `"` characters.

Reply:
>But we need to address the two ! at line 32 and throw errors if any of those cannot be unwrapped.

How would I handle those errors? I would expect errors to be thrown if any of those operations fails.

Reply:
this would be better then `try JSONDecoder().decode(String.self, from: responseData)`